Sources/NetworkProtection/Recovery/Reasserting.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Reasserting.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import NetworkExtension
+
+protocol Reasserting: AnyObject {
+    func startReasserting()
+    func stopReasserting()
+}
+
+extension NEPacketTunnelProvider: Reasserting {
+
+    @MainActor
+    func startReasserting() {
+        reasserting = true
+    }
+
+    @MainActor
+    func stopReasserting() {
+        reasserting = false
+    }
+}
```

Comment:
It seems this isn't really used anywhere.  Can we remove this until it's needed?

Reply:
It is now: https://github.com/duckduckgo/BrowserServicesKit/pull/786/files#r1579297467